app/controllers/ingredients_controller.rb

```diff
@@ -0,0 +1,50 @@
+class IngredientsController < ApplicationController
+  before_action :set_ingredient, only: [:show, :edit, :update]
+
+  def create
+    @ingredient = Ingredient.new(ingredient_params)
+
+    if @ingredient.save
+      redirect_to @ingredient, notice: 'Ingredient was successfully created.'
+    else
+      render :new
+    end
+  end
+
+  def new
+    @ingredient = Ingredient.new
+  end
+
+  def index
+    @ingredients = Ingredient.all
+  end
+
+  def show
+    @ingredient = Ingredient.find(params[:id])
+  end
+
+  def edit
+  end
+
+  def update
+    respond_to do |format|
+      if @ingredient.update(ingredient_params)
+        format.html { redirect_to @ingredient, notice: 'Ingredient was successfully updated.' }
+        format.json { render :show, status: :ok, location: @ingredient }
```

Comment:
Any reason for the `json`?

Reply:
Just going from a previous lab's example, I'll refactor this one and strip it out since its not necessary.

Reply:
It's all good, don't worry about it. As long as you understand why it's not necessary then it's fine to leave it as it.